### ISSUE TITLE
Overriding property with attributes

### ISSUE THREAD

**Issue by mikhail-mmlc** [microsoft/pyright#4445]
**Describe the bug**
pyright doesn't respect attribute's return type if parent has a property with the same name.

**To Reproduce**
```python

class A:
    @property
    def attr(self):
        return 1

class B(A):
    attr = -1  # error: Expression of type "Literal[-1]" cannot be assigned to declared type "property" "Literal[-1]" is incompatible with "property" (reportGeneralTypeIssues)

reveal_type(A().attr)  # Type of "A().attr" is "Literal[1]"
reveal_type(B().attr)  # Type of "B().attr" is "Literal[1]"

```

**Expected behavior**
The linter respects attribute's return type (`Literal[-1]` in the example above).

**VS Code extension or command-line**
`pyright` command line: `pyright example.py`.

**Comment by erictraut** [comment]
This is by design. A property and an attribute have similar, but not the same semantics. It is therefore considered a type violation to override a property with a non-property or vice versa. The type-safe way to do this is to use a property in the subclass.

**Comment by mikhail-mmlc** [comment]
> `It is therefore considered a type violation to override a property with a non-property or vice versa. The type-safe way to do this is to use a property in the subclass.`

and `pyright` rightly prints an error about this. 
However, I don't think that should prevent the linter from relying on the class implementation instead of parent. 
It is also not consistent with non-attribute cases, e.g., when property overrides class method:

```python

class A:

    @property
    def attr(self):
        return -1

    @classmethod
    def method(cls, x: int) -> str:
        return str(x)


class B(A):
    attr = -1  # error: Expression of type "Literal[-1]" cannot be assigned to declared type "property" "Literal[-1]" is incompatible with "property" (reportGeneralTypeIssues)

    static = True

    @property
    def method(self):
        return True

class C(A):
    @staticmethod
    def attr():
        return True


reveal_type(A().attr)  # Type of "A().attr" is "Literal[1]"
reveal_type(B().attr)  # Type of "B().attr" is "Literal[1]"

reveal_type(A().method)  # Type of "A().method" is "(x: int) -> str"
reveal_type(B().method)  # Type of "B().method" is "Literal[True]"

reveal_type(A().attr)  # Type of "A().attr" is "Literal[-1]"
reveal_type(C().attr)  # Type of "C().attr" is "() -> Literal[True]"

```

**Comment by erictraut** [comment]
Pyright is working as designed here.

The issue is that you have declared the type of `attr` in the base class, and then you're attempting to assign an incompatible type to it in a child class. That's a type violation, and it's the job of a type checker to report such type violations.

In the other example, you declared the type of `method` in the base class and redeclared the type in the child class. This redeclaration falls under a different (strict-mode) check called `reportIncompatibleMethodOverride`.

**Comment by mikhail-mmlc** [comment]
Thank you for the clarification.
